src/collectionWatcher.coffee

```diff
@@ -5,27 +5,50 @@ rw.factory 'CollectionWatcher', (ResourceWatcher, $q) ->
 		constructor: (@scope, @collection) ->
 			@resourceWatchers = @collection.map (it) =>
 				@_createResourceWatcher it
+			@hasChanges = false
+			@watchCollection()
 
 		cancel: =>
 			@_removeNewElements @collection
+			@_rollbackCollection()
 			@_rollbackResources()
 
 		_removeNewElements: (collection) =>
 			_.remove collection, (it) => it.isNew()
 
 		save: (options) =>
+			@watchCollection()
+			@_deleteIfNecesary()
 			savePromises = @collection.map _.partial @_saveResource, options
 			$q.all savePromises
+			@hasChanges = false
+
+		_deleteIfNecesary: =>
+			objectsIds = _.map @collection, (it) => it.id
+			toDeleteElements = _.filter @previousState, (it) => !_.includes @collection, it
```

Comment:
Acá en vez de `filter` y adentro *not*, podés usar `_.reject @previousState, (it) => _.includes @collection, it`